### PR TITLE
add more tests for convertor tests

### PR DESCRIPTION
**Pull request by L-M-Sherlock**

### CODE REVIEW

**Comment by dae** [comment]
Thanks Jarrett, those tests look like they'll be quite helpful. I'm a bit short on time today, so it will probably be a day before I can review this.

**Review by dae — APPROVED** [review]
I've ported your tests to Anki, and used some helpers to make the tests a bit easier to read/write. The changes here look fine, and they could either be merged in as-is, or you're welcome to use the helpers if you like.

https://github.com/ankitects/anki/commit/e1b555520cb5058db45f26e167dc5eee65278cb3

File: src/convertor_tests.rs [hunk 1]
```diff
@@ -9,116 +9,153 @@ use itertools::Itertools;
...
+fn convert_to_fsrs_items(
+    mut entries: Vec<RevlogEntry>,
+    next_day_starts_at: i64,
+    timezone: Tz,
+) -> Option<Vec<FSRSItem>> {
+    entries = filter_out_cram(entries);
+    entries = filter_out_set_due_date(entries);
+    entries = remove_revlog_before_forget(entries);
+    entries = remove_revlog_before_last_first_learn(entries);
+    entries = keep_first_revlog_same_date(entries, next_day_starts_at, timezone);
```
[the remark this answers is not in the document]
Comment (L-M-Sherlock) [reply src/convertor_tests.rs:151]: Yeah. I write them in this way just for readability. Feel free to unfold them.

File: src/convertor_tests.rs [hunk 2]
```diff
@@ -478,3 +506,500 @@ fn delta_t_is_correct() -> Result<()> {
...
+            usn: 12394,
+            button_chosen: 3,
+            interval: 131,
+            last_interval: 73,
+            ease_factor: 1750,
+            taken_millis: 30443,
+            review_kind: Review,
+        },
+    ];
+    let revlog_vec = filter_out_cram(revlog_vec);
```
[the remark this answers is not in the document]
Comment (L-M-Sherlock) [reply src/convertor_tests.rs:557]: OK. But it's easier to locate the bug by unit test during development.

[the remark this answers is not in the document]
Comment (dae) [reply src/convertor_tests.rs:557]: Yep, that's true. So long as we're maintaining separate copies of this code in the two projects, maybe your approach makes more sense in this crate.

File: src/convertor_tests.rs [hunk 3]
```diff
@@ -9,116 +9,153 @@ use itertools::Itertools;
...
+
+fn remove_revlog_before_forget(entries: Vec<RevlogEntry>) -> Vec<RevlogEntry> {
+    let mut forget_index = 0;
+    for (index, entry) in entries.iter().enumerate() {
+        if (entry.review_kind == Manual || entry.button_chosen == 0) && entry.ease_factor == 0 {
+            forget_index = index + 1;
+        }
+    }
+    entries[forget_index..].to_vec()
+}
```
[the remark this answers is not in the document]
Comment (L-M-Sherlock) [reply src/convertor_tests.rs:102]: Fine.  The `remove_revlog_before_last_first_learn` function has implemented it, so it's duplicated. But I want to keep it in this place for readability.

**Pull request merged**